File: morgan/python/ch8-insertion-sort.py



def insertionSort(inputs):
    l = len(inputs)
    for i in range(1, l):
        pivotValue = inputs[i]
        toSwapIndex = i
        for j in range(i - 1, -1, -1):
            if inputs[j] > pivotValue:
                inputs[j + 1] = inputs[j]
                toSwapIndex = j
            else:
                break
        inputs[toSwapIndex] = pivotValue



inputs = [10,5,2,1,99]

insertionSort(inputs)

print inputs

File: morgan/php/ch8-insertion-sort.php
<?php


function insertionSort(&$inputs) {
    $l = count($inputs);
    for ($i = 1; $i < $l; $i ++) {
        $pivotValue = $inputs[$i];
        $swapIndex = $i;
        for ($j = $i - 1; $j >= 0; $j --) {
            if ($inputs[$j] > $pivotValue) {
                $swapIndex = $j;
                $inputs[$j + 1] = $inputs[$j];
            }
            else {
                break;
            }
        }
        $inputs[$swapIndex] = $pivotValue;
    }
}


$inputs = [10,5,2,1,99];

insertionSort($inputs);

print_r($inputs);